shell/platform/darwin/macos/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -77,6 +87,60 @@ + (void)respondFalseForSendEvent:(const FlutterKeyEvent&)event
   EXPECT_TRUE(value);
 }
 
+TEST(FlutterViewController, HasViewThatHidesOtherViewsInAccessibility) {
+  FlutterViewController* viewControllerMock = CreateMockViewController(nil);
+
+  [viewControllerMock loadView];
+  auto subViews = [viewControllerMock.view subviews];
+
+  EXPECT_EQ([subViews count], 1u);
+  EXPECT_EQ(subViews[0], viewControllerMock.flutterView);
+
+  NSTextField* textField = [[NSTextField alloc] initWithFrame:NSMakeRect(0, 0, 1, 1)];
+  [viewControllerMock.view addSubview:textField];
+
+  subViews = [viewControllerMock.view subviews];
+  EXPECT_EQ([subViews count], 2u);
+
+  auto accessibilityChildren = viewControllerMock.view.accessibilityChildren;
+  // The accessibilityChildren should only contains the FlutterView.
+  EXPECT_EQ([accessibilityChildren count], 1u);
+  EXPECT_EQ(accessibilityChildren[0], viewControllerMock.flutterView);
+}
+
+TEST(FlutterViewController,
+     MakesFlutterViewFirstResponderWhenAccessibilityIsTurnedOffAndTextInputPluginIsFirstResponder) {
```

Comment:
This might be a bit too long :)

```
SetsFlutterViewFirstResponderWhenAccessibilityDisabled
```
And then just describe the full scenario in a comment above the `TEST()` line.